modal/cli/run.py

```diff
@@ -385,38 +386,47 @@ def run(ctx, write_result, detach, quiet, interactive, env):
     modal run my_app.py::hello_world
     ```
 
-    If your module only has a single app called `app` and your app has a
+    If your module only has a single app and your app has a
     single local entrypoint (or single function), you can omit the app and
     function parts:
 
     ```
     modal run my_app.py
     ```
 
-    Instead of pointing to a file, you can also use the Python module path:
+    Instead of pointing to a file, you can also use the Python module path, which
+    by default will ensure that your remote functions will use the same module
+    names as they do locally.
 
     ```
-    modal run my_project.my_app
+    modal run -m my_project.my_app
     ```
     """
     ctx.ensure_object(dict)
     ctx.obj["result_path"] = write_result
     ctx.obj["detach"] = detach  # if subcommand would be a click command...
     ctx.obj["show_progress"] = False if quiet else True
     ctx.obj["interactive"] = interactive
+    ctx.obj["use_module"] = module
 
 
 def deploy(
-    app_ref: str = typer.Argument(..., help="Path to a Python file with an app."),
+    app_ref: str = typer.Argument(..., help="Path to a Python file with an app to deploy"),
     name: str = typer.Option("", help="Name of the deployment."),
     env: str = ENV_OPTION,
     stream_logs: bool = typer.Option(False, help="Stream logs from the app upon deployment."),
     tag: str = typer.Option("", help="Tag the deployment with a version."),
+    module_mode: bool = typer.Option(False, "--module", "-m", help="Use a Python module path instead of a file path"),
```

Comment:
If we're trying to match Python, I don't think it actually supports `--module`, just `-m`.